tests/routers/test_score_set.py

```diff
@@ -155,6 +191,83 @@ def test_cannot_add_scores_to_other_user_score_set(session, client, setup_router
     assert f"score set with URN '{score_set['urn']}' not found" in response_data["detail"]
 
 
+# If an anonymous user cannot add scores to a data set, they won't be able to add scores and counts.
+def test_anonymous_cannot_add_scores_to_other_user_score_set(
+    session, client, setup_router_db, data_files, anonymous_app_overrides
+):
+    experiment = create_experiment(client)
+    score_set = create_seq_score_set(client, experiment["urn"])
+    change_ownership(session, score_set["urn"], ScoreSetDbModel)
+    scores_csv_path = data_files / "scores.csv"
+
+    with open(scores_csv_path, "rb") as scores_file, DependencyOverrider(anonymous_app_overrides):
+        response = client.post(
+            f"/api/v1/score-sets/{score_set['urn']}/variants/data",
+            files={"scores_file": (scores_csv_path.name, scores_file, "text/csv")},
+        )
+
+    assert response.status_code == 401
+    response_data = response.json()
+    assert "Could not validate credentials" in response_data["detail"]
+
+
+def test_admin_can_add_scores_to_other_user_score_set(
```

Comment:
The intent is to ensure that the endpoint accepts the counts file as optional and you can either supply the scores or the counts and scores. It might not be necessary.